base/dcp_client_test.go

```diff
@@ -393,17 +390,36 @@ func TestContinuousDCPRollback(t *testing.T) {
 	select {
 	case <-c:
 		mutationCount := atomic.LoadUint64(&mutationCount)
-		require.Equal(t, uint64(1000), mutationCount)
+		require.Equal(t, uint64(10000), mutationCount)
 	case <-timeout:
 		t.Fatalf("timeout on client reached")
 	}
 
+	// new dcp client to simulate a rollback
+	dcpClientOpts = DCPClientOptions{
+		InitialMetadata:   dcpClient.GetMetadata(),
+		FailOnRollback:    false,
+		OneShot:           false,
+		CollectionIDs:     collectionIDs,
+		CheckpointPrefix:  DefaultMetadataKeys.DCPCheckpointPrefix(t.Name()),
+		MetadataStoreType: DCPMetadataStoreInMemory,
+	}
+	require.NoError(t, dcpClient.Close())
+
+	dcpClient1, err := NewDCPClient(feedID, counterCallback, dcpClientOpts, gocbv2Bucket)
+	require.NoError(t, err)
+	// function to force the rollback of some vBuckets
+	dcpClient1.forceRollbackvBucket(vbUUID)
+
+	_, startErr = dcpClient1.Start()
+	require.NoError(t, startErr)
+
 	// Assert that the number of vBuckets active are the same as the total number of vBuckets on the client.
 	// In continuous rollback the streams should not close after they're finished.
-	numVBuckets := len(dcpClient.activeVbuckets)
-	require.Equal(t, dcpClient.numVbuckets, uint16(numVBuckets))
+	numVBuckets := len(dcpClient1.activeVbuckets)
+	require.Equal(t, dcpClient1.numVbuckets, uint16(numVBuckets))
 
-	require.NoError(t, dcpClient.Close())
+	require.NoError(t, dcpClient1.Close())
```

Comment:
I'd move this to a

```
defer func() {
    assert.NoError(t, dcpClient1.Close())
}
```

an assert because in a defer we don't want to change this